flambe/metric/metric.py

```diff
@@ -32,6 +34,55 @@ def compute(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         """
         pass
 
+    def aggregate(self, state: dict, *args, **kwargs) -> Dict:
+        """
+
+        Parameters
+        ----------
+        state: dict
+            the state dictionary
+        args:
+            normally pred, target
+        kwargs
+
+        Returns
+        -------
+        The updated state (even though the update happens in-place)
+        """
+        score = self.compute(*args, **kwargs)
+        if isinstance(score, torch.Tensor):
+            score_np = score.cpu().detach().numpy()
+        else:
+            score_np = score
+        try:
+            num_samples = args[0].size(0)
+        except (ValueError, AttributeError):
+            raise ValueError(f'Cannot get size from {type(args[0])}')
+        if state == {}:
```

Comment:
ditto on the defaultdict comment above, and simplifying the declaration if not

Reply:
as above,  modification has to happen in-place

Reply:
you can check emptyness with `if not state`